### PR TITLE
Add dynamic quant to the tutorial of PT2E quantization with X86Inductor

### PR DESCRIPTION
**Pull request by Xia-Weiwen**
## Description

Add descriptions about dynamic quantization in the tutorial of PT2E quantization with X86Inductor. Dynamic quantization with X86Inductor will be a new feature in PyTorch 2.3.

## Checklist

- [ ] The issue that is being fixed is referred in the description (see above "Fixes #ISSUE_NUMBER")
- [ ] Only one issue is addressed in this pull request
- [ ] Labels from the issue that this PR is fixing are added to this pull request
- [x] No unnecessary issues are included into this pull request.

### CODE REVIEW

File: prototype_source/pt2e_quant_x86_inductor.rst [hunk 1]
```diff
@@ -20,6 +20,8 @@ for the x86 inductor backend and explains how to lower the quantized model into
 The pytorch 2 export quantization flow uses the torch.export to capture the model into a graph and perform quantization transformations on top of the ATen graph.
 This approach is expected to have significantly higher model coverage, better programmability, and a simplified UX.
 TorchInductor is the new compiler backend that compiles the FX Graphs generated by TorchDynamo into optimized C++/Triton kernels.
+It supports both static and dynamic quantization. For the different between the two quantization types, please refer to the doc
```
Comment (jgong5) [inline prototype_source/pt2e_quant_x86_inductor.rst:23]:
```suggestion
It supports both static and dynamic quantization. For the difference between the two quantization types, please refer to the doc
```

**Review by jgong5 — CHANGES_REQUESTED** [review]
Can we also explain how the models would be quantized with the dynamic quantization mode?

File: prototype_source/pt2e_quant_x86_inductor.rst [hunk 2]
```diff
@@ -21,7 +21,11 @@ The pytorch 2 export quantization flow uses the torch.export to capture the mode
 This approach is expected to have significantly higher model coverage, better programmability, and a simplified UX.
 TorchInductor is the new compiler backend that compiles the FX Graphs generated by TorchDynamo into optimized C++/Triton kernels.
 
-This flow of quantization 2 with Inductor mainly includes three steps:
+This flow of quantization 2 with Inductor supports both static and dynamic quantization. Static quantization works best for CNN models, like ResNet-50. And dynamic quantization is more suitable for NLP models, like RNN and BERT.
+For the difference between the two quantization types, please refer to the doc
+`here <https://pytorch.org/docs/stable/quantization.html#quantization-mode-support>`_.
```
Comment (svekars) [inline prototype_source/pt2e_quant_x86_inductor.rst:26]:
```suggestion
For the difference between the two quantization types, please refer to the `following page <https://pytorch.org/docs/stable/quantization.html#quantization-mode-support>`__.
```

File: prototype_source/pt2e_quant_x86_inductor.rst [hunk 3]
```diff
@@ -134,14 +138,22 @@ quantize the model.
   `multiplications are 7-bit x 8-bit <https://oneapi-src.github.io/oneDNN/dev_guide_int8_computations.html#inputs-of-mixed-type-u8-and-s8>`_. In other words, potential
   numeric saturation and accuracy issue may happen when running on CPU without Vector Neural Network Instruction.
 
+The quantization config is for static quantization by default. To apply dynamic quantization, add an argument ``is_dynamic=True`` when getting the config.
+
+::
```
Comment (svekars) [inline prototype_source/pt2e_quant_x86_inductor.rst:143]: please use code-block directives instead of ::. 
```suggestion
.. code-block:: python
```

Comment (Xia-Weiwen) [reply prototype_source/pt2e_quant_x86_inductor.rst]: Thanks.

**Comment by Xia-Weiwen** [comment]
Hi @svekars Do you have more comments? Thanks

**Comment by svekars** [comment]
@jgong5  if this looks good to you, can you please approve?

**Review by svekars — APPROVED** [review]
